magley/klau-drive
# Feat sharing

Closes #25 #26 and partially (80-90%-ish) of #30 

How to test:
register 2 users (e.g. `bob` and `john`)
`bob` uploads stuff (files and nested albums)
`bob` shares root-level file with `john`
`bob` shares root-level nested folder with `john`
`bob` shares root-level file inside nested folder with `john`
login as `john` and confirm you can access these items and download them
`bob` stops sharing root-level file with `john` (2rd tab, you have to enter the username to get shared items)
`bob deletes root-level nested folder (should stop sharing both the folder and the root-level file)
`john` doesn't have access to anything anymore

```
bob_root:
    file1.txt      <- share this
    folder1        <- this
      folder2
        file2.txt  <- and this
```

Things I took care of:
- can't upload or add albums inside a shared album (i.e. you have to be the owner)
- can't delete, move, update shared content
- can't list items that you don't own/isn't shared with you
Many of these are hard to check on the frontend because we don't allow it by design, but the server doesn't allow it either.
